comp110-worksheet-6/OxoBoard.cs

```diff
@@ -1,47 +1,122 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace comp110_worksheet_6
 {
-	public enum Mark { None, O, X };
+public enum Mark { None, O, X };
+ 
 
-	public class OxoBoard
-	{
-		// Constructor. Perform any necessary data initialisation here.
-		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+    public class OxoBoard
+{
+        public int width = 3;
+        public int height = 3;
+        public int inARow = 3;
+        int[,] myGrid = new int[width, height];
+        public bool isBoardFull;
+        public int topLeft;
+        public int topMiddle;
+
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            GetSquare();
+            SetSquare();
+            IsBoardFull();
+            GetWinner();
 		}
 
 		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+	public Mark GetSquare(int x, int y)
+	{
+            if(x == 0 && y == 0)
+            {
+                return topLeft;
+            }
```

Comment:
This approach of having a separate variable per square would quickly get unwieldy (as I think you may have discovered) -- you have a 2d array `myGrid`, this would be much better to use to store the state of the board